### PR TITLE
Date range request mapping

### PR DESCRIPTION
**Pull request by valnoel**

### CODE REVIEW

File: lib/ex_video_factory.ex [hunk 1]
```diff
@@ -15,6 +15,7 @@ defmodule ExVideoFactory do
       Map.get(query, "per_page", 20)
       |> force_integer
     query = format_channels(query)
+    query = format_dates(query)
```
[the remark this answers is not in the document]
Comment (valnoel) [reply lib/ex_video_factory.ex]: 👍 See: b046ddc

File: lib/ex_video_factory.ex [hunk 2]
```diff
@@ -99,48 +99,26 @@ defmodule ExVideoFactory do
...
-
-  defp format_dates(%{"produced_after" => start_date, "produced_before" => end_date} = query) do
+  defp format_dates(query) do
     query
-    |> Map.put("produced_at[gte]", format_iso8601(start_date))
-    |> Map.put("produced_at[lte]", format_iso8601(end_date))
-    |> Map.delete("produced_after")
-    |> Map.delete("produced_before")
+    |> Enum.map(fn {key, value} ->
+        cond do
```
[the remark this answers is not in the document]
Comment (valnoel) [reply lib/ex_video_factory.ex]: See: 4998fac

File: test/ex_video_factory_test.exs [hunk 3]
```diff
@@ -43,64 +43,19 @@ defmodule ExVideoFactoryTest do
...
-    content = ExVideoFactory.videos(%{"produced_after" => "2018-01-30T12:30:00+01:00", "produced_before" => "2018-01-30T13:30:00+01:00"})
-    assert length(content.videos) == content.size
-    assert content.offset + content.size <= content.total
-    assert content.total >= content.size
-    assert content.total > 0
-  end
-
-  test "get some videos, filtered by update date" do
-    content = ExVideoFactory.videos(%{"updated_after" => "2018-01-30T12:30:00+01:00", "updated_before" => "2018-01-30T13:30:00+01:00"})
+  test "get some videos, filtered by dates (and other)" do
```
[the remark this answers is not in the document]
Comment (valnoel) [reply test/ex_video_factory_test.exs]: Yes, done into 2b818ed

File: lib/ex_video_factory.ex [hunk 4]
```diff
@@ -99,26 +101,26 @@ defmodule ExVideoFactory do
...
-  defp format_date_key(key) do
-    key
-    |> String.replace("_after", "_at[gte]")
-    |> String.replace("_before", "_at[lte]")
+  defp format_field(_, [prefix, "after"], value) do
+    {prefix <> "_at[gte]", format_iso8601(value)}
+  end
+  defp format_field(_, [prefix, "before"], value) do
+    {prefix <> "_at[lte]", format_iso8601(value)}
+  end
```
Comment (MarcAntoine-Arnaud) [inline lib/ex_video_factory.ex]: is it possible to change: `format_channels` per `format_query`

and add here
```
defp format_field("channels[]", _, value) do
   do stuff here
end
```

**Pull request merged**